pygama/genpar_tmp/optimiser_macro.py

```diff
@@ -0,0 +1,52 @@
+import numpy as np
+import os,json
+from pygama.analysis import histograms as hist
+import pygama.lh5 as lh5
+import pygama.dsp.dsp_optimize as opt
+#import pygama.genpar_tmp.cuts as cuts
+sto = lh5.Store()
+
+def run_optimisation(file,opt_config,dsp_config,fom,cuts):
+    grid = set_par_space(list(opt_config.keys())[0],opt_config[list(opt_config.keys())[0]],init_args = True)
+    waveforms = sto.read_object('/raw/waveform', file,idx=cuts,verbosity=0)[0]
+    tb_data = lh5.Table(col_dict = { 'waveform' : waveforms } )
+    return opt.run_grid(tb_data,dsp_config,grid,fom,verbosity=0)
+
+def set_par_space(processor, par_values,init_args=False):
+    par_space = opt.ParGrid()
+    for ind in par_values.keys():
+        str_vals = set_values(par_values[ind])
+        par_space.add_dimension(processor,int(ind),str_vals,init_arg = init_args)
+    return par_space
+    
+def set_values(par_values):
+    string_values=np.linspace(par_values['start'],par_values['end'],par_values['frequency'])
+    string_values = [ f'{val:.2f}{par_values["unit"]}' for val in string_values]
+    return string_values
+
+def build_energy_cuts(file,Emin,Emax):
```

Comment:
I think it would be better to change this function to output a mask as this will be much easier to combine with other cuts (also simplifies the code) something like mask = (wf_maxes>Emin)&(wf_maxes<Emax)

Reply:
Changed